client/components/data/domain-management/email/index.jsx

```diff
@@ -22,110 +22,117 @@ import {
 import { getPlansBySite } from 'state/sites/plans/selectors';
+import { getProductsList } from 'state/products-list/selectors';
+import { getSelectedSite } from 'state/ui/selectors';
 
-const user = userFactory();
-
-var stores = [
-	DomainsStore,
+const stores = [
 	CartStore
 ];
 
 function getStateFromStores( props ) {
 	return {
-		domains: DomainsStore.getBySite( props.selectedSite.ID ) || {},
+		domains: props.domains,
 		cart: CartStore.get(),
 		context: props.context,
 		products: props.products,
 		selectedDomainName: props.selectedDomainName,
 		selectedSite: props.selectedSite,
 		sitePlans: props.sitePlans,
-		user: user.get(),
+		user: props.user,
 		googleAppsUsers: props.googleAppsUsers,
 		googleAppsUsersLoaded: props.googleAppsUsersLoaded
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
-
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		context: React.PropTypes.object.isRequired,
-		productsList: React.PropTypes.object.isRequired,
-		selectedDomainName: React.PropTypes.string,
-		sitePlans: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired,
-		googleAppsUsers: React.PropTypes.array.isRequired,
-		googleAppsUsersLoaded: React.PropTypes.bool.isRequired
-	},
-
-	mixins: [ observe( 'productsList' ) ],
+export class EmailData extends Component {
+	static propTypes = {
+		component: PropTypes.func.isRequired,
+		context: PropTypes.object.isRequired,
+		products: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string,
+		selectedSite: PropTypes.object.isRequired,
+		sitePlans: PropTypes.object.isRequired,
+		googleAppsUsers: PropTypes.array.isRequired,
+		googleAppsUsersLoaded: PropTypes.bool.isRequired,
+	};
 
-	componentWillMount() {
-		this.loadDomainsAndSitePlans();
+	componentDidMount() {
 		this.props.fetchGoogleAppsUsers();
-	},
-
-	componentWillUpdate() {
-		this.loadDomainsAndSitePlans();
-	},
-
-	loadDomainsAndSitePlans() {
-		const selectedSite = this.props.sites.getSelectedSite();
-
-		if ( this.prevSelectedSite !== selectedSite ) {
-			fetchDomains( selectedSite.ID );
-			this.props.fetchSitePlans( this.props.sitePlans, this.props.sites.getSelectedSite() );
-
-			this.prevSelectedSite = selectedSite;
-		}
-	},
+	}
 
 	render() {
+		const {
+			selectedSite: { ID: siteId },
+		} = this.props;
+
 		return (
-			<StoreConnection
-				domains={ this.props.domains }
-				googleAppsUsers={ this.props.googleAppsUsers }
-				googleAppsUsersLoaded={ this.props.googleAppsUsersLoaded }
-				component={ this.props.component }
-				stores={ stores }
-				getStateFromStores={ getStateFromStores }
-				products={ this.props.productsList.get() }
-				selectedDomainName={ this.props.selectedDomainName }
-				selectedSite={ this.props.sites.getSelectedSite() }
-				sitePlans={ this.props.sitePlans }
-				context={ this.props.context } />
+			<div>
+				<QueryProducts />
+				<QuerySiteDomains { ...{ siteId } } />
```

Comment:
this is a pattern I have used for a long time to eliminate needless redundancies in the code. writing something like `siteId={ siteId }` bugs me, especially when it's like five or more values all of the same name as the attribute